Poet client to register with Validator Registry

arsulegai opened the pull request:
IAS client / proxy is not merged yet to the main branch. Submitting 2 commits on PoET2 client along with IAS part because of dependency. Please review only two of the commits.

rranjan3 commented on src/ias_proxy/src/lib.rs:
```diff
@@ -0,0 +1,37 @@
+*/
+
+#[macro_use]
+extern crate serde_derive;
```
Move crate declarations to main/mod/lib

arsulegai replied:
I didn't understand, do you mean move lib.rs as a module in main?

rranjan3 commented on src/ias_proxy/src/ias_proxy_server/mod.rs:
```diff
@@ -0,0 +1,209 @@
+                send_response(404, None, None)
+            },
+        Method::POST =>
+            if req.uri().path().contains("/attestation/sgx/v2/report") {
```
Move this to some constant/config

arsulegai replied:
Ok

rranjan3 commented on src/ias_proxy/src/ias_proxy_server/mod.rs:
```diff
@@ -0,0 +1,209 @@
+    let response = match *req.method() {
+        // handle get request from the proxy
+        Method::GET =>
+            if path.contains("/attestation/sgx/v2/sigrl") {
```
Use a constant.

arsulegai replied:
Ok

rranjan3 commented on src/ias_proxy/src/ias_proxy_server/mod.rs:
```diff
@@ -0,0 +1,209 @@
+        };
+        let server = Server::bind(&socket_addr).serve(new_service);
+        hyper::rt::run(server.map_err(|e| {
+            panic!("Server error: {}", e);
```
Is **panic** really pertinent whenever used? Can logging an error message suffice?

arsulegai replied:
I didn't know idiomatic way of error handling in Rust. Please suggest a reference link, will also talk to you on this.

rranjan3 commented on src/ias_proxy/src/ias_proxy_server/mod.rs:
```diff
@@ -0,0 +1,209 @@
+        IasProxyServer {
+            ias_proxy_name: config_map["proxy_name"].clone(),
+            ias_proxy_port: config_map["proxy_port"].clone(),
+            ias_client: IasClient::new(config_map["ias_url"].clone(), Vec::from(config_map["spid_cert_file"].as_bytes()), None),
```
Wrap this around to ~80 chars.

arsulegai replied:
Ok

rranjan3 commented on src/ias_client/src/client_utils.rs:
```diff
@@ -0,0 +1,83 @@
+}
+
+/// Return future of body from ResponseFuture
+pub fn read_body_as_string_from_response(response: ResponseFuture, header_param_to_read: Option<&'static str>) -> impl Future<Item=String, Error=Error> {
```
Wrap

arsulegai replied:
Ok

arsulegai commented on .gitignore:
```diff
@@ -27,3 +27,10 @@
 
 *.batch
+
+src/ias_client/target/**
```
Ok

arsulegai commented on demo-sgx.yaml:
```diff
@@ -21,6 +21,18 @@ volumes:
+        - https_proxy
+        - no_proxy
+    working_dir: /project/ias_proxy/target/debug/
+    command: "ias_proxy -c dummy-config.toml -vvv"
```
Ok, good catch

arsulegai commented on src/core/src/poet2_util.rs:
```diff
@@ -1,11 +1,40 @@
     let strs: Vec<String> = bytes.iter()
         .map(|b| format!("{:02x}", b))
-    .collect();
+        .collect();
```
I think it's editor auto-indenting. Will force this part.

arsulegai commented:
Addressed most of these comments and created a new commit because these files were from multiple commits. One is to move all constants to separate file (there were other constants too, will move them together in refactoring or UT patch), another is to refrain from IDE made indentation changes (I disabled this feature for that one particular file, it's taking more time to get back previous state).

rranjan3 commented on src/ias_proxy/src/lru_cache/mod.rs:
```diff
@@ -0,0 +1,74 @@
+    // Size of the LRU cache
+    max_size: usize,
+    // A list to note which key is accessed first, it should be locked before accessing
+    // order: Mutex<VecDeque<String>>,
```
Do away with dead code.

arsulegai replied:
Yes, I have cleaned up these unused code and improved error handling (not using panic) in unit test patch.

rranjan3 commented on src/ias_proxy/src/lru_cache/mod.rs:
```diff
@@ -0,0 +1,74 @@
+        let size = size.unwrap_or(100);
+        LruCache {
+            max_size: size,
+            // order: Mutex::new(VecDeque::with_capacity(size)),
```
dead code

arsulegai replied:
Cleaned up dead code in unit test cases patch.

rranjan3 commented on src/ias_proxy/src/ias_proxy_server/mod.rs:
```diff
@@ -0,0 +1,216 @@
+
+    /// stop to stop listening on the port
+    pub fn stop(&self) -> () {
+        // TODO: Need to stop the server started
```
Missed some implementation or deliberate?

arsulegai replied:
It's deliberate, still couldn't figure out a way to stop the server. I was trying similar logic in unit test patch which is much smaller and simpler compared to this but wasn't successful. Will leave this as TODO until it's resolved. Problem is hyper doesn't have a working example which we want for the version we're using. Either information available online on hyper is outdated to older versions or there's no example available for the latest version. Available solutions didn't work for the version we're using. Tried few other alternate way of doing away but am not successful yet.

askmish commented:
Please specify which commits to review?
